src/TurboMqtt/Client/ClientStreamOwner.cs

```diff
@@ -336,6 +335,19 @@ private void Running(object message)
                     var requiredActors = new MqttRequiredActors(_exactlyOnceActor!, _atLeastOnceActor!,
                         _clientAckActor!,
                         _heartBeatActor!);
+                    
+                    // the transport should be at rest now, no longer being written to - clear out all the old data\
+                    HashSet<MqttPacket> preservedPackets = new();
+                    while (_outboundChannel!.Reader.TryRead(out var p))
```

Comment:
Fully drain the outbound channel before we hook it back up to the new `IMqttTransport` - at this point writing to this channel should be paused by the changes we made to `MqttClient.PublishAsync`